tools/travis-before-install.sh

```diff
@@ -36,6 +36,6 @@ fi
 
 
 pip install --upgrade pip setuptools
-pip install pytz cython pytest==5.0.1
+pip install pytz cython pytest
```

Comment:
I still think pinning is preferable, and we can periodically issue a PR to update the version in a controlled way

Reply:
The problem is, we don't know of any working versions of pytest :) I've opened a task issue to fix that when there is a working version.

For the NumPy 1.16.x and 1.17.x branches I will pin the pytest version to 5.0.1 since we know that works with python3.5-dbg. Or at least the problem goes unreported

Reply:
My fear is that unless we pin everything in one place, we will forget we ever pinned it.

Reply:
I have a task for that with 1.18.0 milestone, see #14293.